api/src/test/java/jakarta/data/displaynamegeneration/test/ReplaceCamelCaseAndUnderscoreAndNumberTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022,2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.displaynamegeneration.test;
+
+import jakarta.data.displaynamegeneration.ReplaceCamelCaseAndUnderscoreAndNumber;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+//@DisplayNameGeneration(ReplaceCamelCaseAndUnderscoreAndNumber.class)
+class ReplaceCamelCaseAndUnderscoreAndNumberTest {
+
+    private static Stream<Arguments> provideMethodNameAndExpectedDisplayName() {
+        return Stream.of(
+                Arguments.of("shouldReturnErrorWhen_maxResults_IsNegative", "Should return error when maxResults is negative"),
+                Arguments.of("shouldCreateLimitWithRange", "Should create limit with range"),
+                Arguments.of("shouldReturn5Errors", "Should return 5 errors"),
+                Arguments.of("shouldReturn5errors", "Should return 5errors"),
+                Arguments.of("shouldReturn23Errors", "Should return 23 errors"),
+                Arguments.of("shouldReturnTheValueOf_maxResults", "Should return the value of maxResults"),
+                Arguments.of("shouldReturnTheNumberOfErrorsAs_numberOfErrors_InferiorOrEqualTo5", "Should return the number of errors as numberOfErrors inferior or equal to 5"),
+                Arguments.of("shouldReturnTheNumberOfErrorsAs_numberOfErrors_InferiorOrEqualTo15", "Should return the number of errors as numberOfErrors inferior or equal to 15")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideMethodNameAndExpectedDisplayName")
+    void shouldReturnMethodDisplayNamesForCamelCaseAndUnderscoreAndNumber(final String methodName, final String expectedDisplayName) {
+        assertSoftly(softly -> {
+            try {
+                softly.assertThat(ReplaceCamelCaseAndUnderscoreAndNumber.INSTANCE.generateDisplayNameForMethod(getClass(), getClass().getDeclaredMethod(methodName))).isEqualTo(expectedDisplayName);
+            } catch (NoSuchMethodException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
```

Comment:
Do these methods need to be annotated with `@Test`? I don't think that the `ReplaceCamelCaseAndUnderscoreAndNumber` actually checks that the method passed in is annotated with `@Test`

Having these annotated will just clog the test results.

Reply:
Sure, these methods must be annotated with @Test or @ParameterizedTest. I don't think that we need to check if this method is annotated correctly because by default this generator will only focus on test methods. 
As you can see, the parent class DisplayNameGenerator.Standard does not check anything

```java
		@Override
		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
			String displayName = testMethod.getName();
			if (hasParameters(testMethod)) {
				displayName += ' ' + parameterTypesAsString(testMethod);
			}
			return displayName;
		}
```

Then, ReplaceCamelCaseAndUnderscoreAndNumberTest must check only the method annotated correctly. I will update